File: java/hcx-integrator-sdk/src/main/java/io/hcxprotocol/dto/NotificationRequest.java
package io.hcxprotocol.dto;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.typesafe.config.Config;
import io.hcxprotocol.utils.JSONUtils;

import java.util.List;
import java.util.Map;

public class NotificationRequest {
    private String jwsPayload;
    private String topicCode;
    private List<String> recipients;
    private String recipientType;
    private String message;
    private String correlationId;
    private Map<String,String> templateParams;

    private Config config;
    public NotificationRequest(String jwsPayload){
        this.jwsPayload = jwsPayload;
    }

    public NotificationRequest(String topicCode, String message, Map<String, String> templateParams, String recipientType, List<String> recipients,String correlationId,Config config) {
        this.topicCode = topicCode;
        this.message = message;
        this.templateParams = templateParams;
        this.recipientType = recipientType;
        this.recipients = recipients;
        this.correlationId = correlationId;
        this.config = config;
    }

    public String getJwsPayload() {
        return jwsPayload;
    }

    public Map<String,Object> getHeaders() throws JsonProcessingException {
       return JSONUtils.decodeBase64String(jwsPayload.split("\\.")[0], Map.class);
    }

    public Map<String,Object> getPayload() throws JsonProcessingException {
        return JSONUtils.decodeBase64String(jwsPayload.split("\\.")[1], Map.class);
    }
    public String getTopicCode() {
        return topicCode;
    }
    public List<String> getRecipients() {
        return recipients;
    }
    public String getRecipientType() {
        return recipientType;
    }

    public String getMessage() {
        return message;
    }
    public Map<String, String> getTemplateParams() {
        return templateParams;
    }

    public String getCorrelationId() {
        return correlationId;
    }

    public Config getConfig() {
        return config;
    }

}

File: dot-net/hcx-integrator-sdk/Io.HcxProtocol/Dto/NotificationRequest.cs
using Io.HcxProtocol.Init;
using Io.HcxProtocol.Utils;
using System.Collections.Generic;

namespace Io.HcxProtocol.Dto
{
    public class NotificationRequest
    {
        private string jwsPayload;
        private string topicCode;
        private List<string> recipients;
        private string recipientType;
        private string message;
        private string correlationId;
        private Dictionary<string, string> templateParams;
        private Config config;

        public NotificationRequest(string jwsPayload)
        {
            this.jwsPayload = jwsPayload;
        }

        public NotificationRequest(string topicCode, string message, Dictionary<string, string> templateParams, string recipientType, List<string> recipients, string correlationId, Config config)
        {
            this.topicCode = topicCode;
            this.message = message;
            this.templateParams = templateParams;
            this.recipientType = recipientType;
            this.recipients = recipients;
            this.correlationId = correlationId;
            this.config = config;
        }

        public string GetJwsPayload()
        {
            return jwsPayload;
        }

        public Dictionary<string, object> GetHeaders()
        {
            var x = jwsPayload.Split('.');
            var j = JSONUtils.DecodeBase64String<Dictionary<string, object>>(jwsPayload.Split('.')[0].ToString());

            return j;
        }

        public Dictionary<string, object> GetPayload()
        {
            return JSONUtils.DecodeBase64String<Dictionary<string, object>>(jwsPayload.Split('.')[0].ToString());
        }

        public string GetTopicCode()
        {
            return topicCode;
        }

        public List<string> GetRecipients()
        {
            return recipients;
        }

        public string GetRecipientType()
        {
            return recipientType;
        }

        public string GetMessage()
        {
            return message;
        }

        public Dictionary<string, string> GetTemplateParams()
        {
            return templateParams;
        }

        public string GetCorrelationId()
        {
            return correlationId;
        }

        public Config GetConfig()
        {
            return config;
        }

        public Dictionary<string, object> NotificationHeaders()
        {
            var x = GetHeaders()[Constants.NOTIFICATION_HEADERS];
            var notificationheaders = JSONUtils.Deserialize<Dictionary<string, object>>(x.ToString());
            return notificationheaders;
        }

        public string GetSenderCode()
        {
            return (string)NotificationHeaders()["sender_code"];
        }
    }
}
